lib/streamlit/cli.py

```diff
@@ -96,12 +96,30 @@ def main_hello():
 
 
 @main.command('run')
-@click.argument('file', type=click.Path(exists=True))
+@click.argument('file', type=click.Path(exists=True), required=False)
 @click.argument('args', nargs=-1)
-def main_run(file, args):
-    """Run a Python script, piping stderr to Streamlit."""
+@click.option('-m', '--module', 'module_name', help='Run module')
+def main_run(file, args, module_name):
```

Comment:
How does `-m` work with `args`? People should be able to pass args using `streamlit run -m foo arg1 arg2`. Is this supported?

Reply:
Good catch. I need a solution for this. It won't work out of the box:
```run -m foo arg1 arg2
...cannot find FILE arg1```
arg1 it is interpreted as the first argument. Looking at a solution

Reply:
One solution could be to just have variadic args and depending on the the context (the `-m` option) interpret the first argument as `file` or as `argument`. I don't like this, because we loose "click validation: for files. This is probably how the python `-m` works.

Another solution is to have invocation for the case of `-m` with options within quotes:
```
run -m "foo arg1 arg2"
```

We will need to do some manual splitting and validation and the quotes are not nice.

Last option is to use `-m` as a boolean switch which tells whether to interpret the first argument as file or module... still I don't like this much, we loose click validation as well and it is not the way python does.

Reply:
I went for the last proposal: `-m` is a boolean switch...